asmcli/components/ca/managed-cas.sh

```diff
@@ -88,9 +99,9 @@ x_wait_for_enabling_workload_certificates() {
   # Enabling workload certificate feature on a cluster will result in
   # the cluster restarting and the cluster becoming unreachable for minutes.
   # The cluster restart may happen at a random time after the enablement.
-  # Check the connection to k8s is recovered.
-  # Therefore, wait 15 minutes before checking the connection to k8s is recovered.
-  sleep 900
+  # The time to complete a cluster varies based on cluster region and size.
+  # Therefore, wait 25 minutes before checking the connection to k8s is recovered.
+  sleep 1500
```

Comment:
1500 second is 25 minutes

I would suggest to revert the sleep time to 900. For a small cluster with few nodes, sleeping 1500 is too long and unnecessary. And for a larger cluster with 20+ nodes, 1500 is not enough. It is okay to sleep 900 and notify customer to rerun the command.

Another option is to add a for loop like you added at L81. In the for loop, check the zatar controller deployment, if it is not existed, sleep for 200 seconds and retry. We can loop for 6+ iterations.

Reply:
I agree with you. The sleep time is reverted to 900.